Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateCSSPropertyID.cpp

```diff
@@ -1159,14 +1180,50 @@ Vector<PropertyID> shorthands_for_longhand(PropertyID property_id)
         property_generator.set("shorthands", builder.to_byte_string());
         property_generator.append(R"~~~(
     case PropertyID::@name:titlecase@:
-        return { @shorthands@ };
+        immediate_shorthands = { @shorthands@ };
+        break;
 )~~~");
     }
 
     generator.append(R"~~~(
-        default:
-            return { };
-        }
+    default:
+        return { };
+    }
+
+    Vector<PropertyID> shorthands;
+
+    for (auto immediate_shorthand : immediate_shorthands) {
+        shorthands.append(immediate_shorthand);
+
+        if (property_maps_to_shorthand(immediate_shorthand))
+            shorthands.extend(shorthands_for_longhand(immediate_shorthand));
+    }
+
+    // https://www.w3.org/TR/cssom/#concept-shorthands-preferred-order
+    AK::quick_sort(shorthands, [&](PropertyID a, PropertyID b) {
```

Comment:
Good catch with this sorting. But can we do all of this inside the code generator instead of at run time? It looks heavy and it's also constant, so we should be able to just return a fixed list like before.

Also a note about why the steps are in a different order than the spec would be helpful.

Reply:
i wrote that, because i believe that the code is incorrect for this more complex case (and i’d like you to prove me wrong), more specifically: steps 2. and 3. are suspicious. maybe it is possible to conjure a property? 🤔